op-service/txmgr/txmgr.go

```diff
@@ -69,6 +80,9 @@ type TxManager interface {
 	//
 	// NOTE: Send should be called by AT MOST one caller at a time.
 	Send(ctx context.Context, tx *types.Transaction) (*types.Receipt, error)
+
+	// CraftTx is used to craft a transaction using a [TxCandidate].
+	CraftTx(ctx context.Context, candidate TxCandidate) (*types.Transaction, error)
```

Comment:
I guess we need `CraftTx` because we're implementing the proposer changes separately, but the goal is that `Send` takes the `TxCandidate` instead of the `Tx`

Reply:
Got it - op-proposer changes (along with the config variables you noted above are in the linear ticket https://linear.app/optimism/issue/CLI-3685/featop-proposer-migrate-the-proposer-to-txmgr-tx-creation so we track). CLI-3685 (the op-proposer changes) will be the next pr I focus on tomorrow after this.